agent/hcp/telemetry_provider.go

```diff
@@ -30,19 +37,25 @@ var (
 // Ensure hcpProviderImpl implements telemetry provider interfaces.
 var _ telemetry.ConfigProvider = &hcpProviderImpl{}
 var _ telemetry.EndpointProvider = &hcpProviderImpl{}
+var _ telemetry.ClientProvider = &hcpProviderImpl{}
 
 // hcpProviderImpl holds telemetry configuration and settings for continuous fetch of new config from HCP.
 // it updates configuration, if changes are detected.
 type hcpProviderImpl struct {
 	// cfg holds configuration that can be dynamically updated.
 	cfg *dynamicConfig
+	// httpCfg holds configuration for the HTTP client
+	httpCfg *httpCfg
 
 	// A reader-writer mutex is used as the provider is read heavy.
 	// OTEL components access telemetryConfig during metrics collection and export (read).
 	// Meanwhile, config is only updated when there are changes (write).
 	rw sync.RWMutex
 	// hcpClient is an authenticated client used to make HTTP requests to HCP.
 	hcpClient client.Client
+
+	// logger is the HCP logger for the provider
+	logger hclog.Logger
```

Comment:
nit: logger from ctx based on previous convos with team members is preferred

Reply:
I ran into issues when trying to add back the `hclog.FromContext(ctx)` because the manager's context is a `StopChannelContext`, which returns a default logger. That default logger then doesn't have the consul formatting or the log configuration.

I could pass a logger to the methods each time, but that feels a little excessive. It would also just pass the HCP manager's logger that's [stored on a struct](https://github.com/hashicorp/consul/blob/main/agent/hcp/manager.go#L32) anyway, so doesn't feel like it actually is an improvement.

I think our only option for now is to keep it on the struct then. We'll be making more changes to the HCP manager in the future, so if we eventually call these methods with a more standard context, we can try switching back to `hclog.FromContext(ctx)`.

Reply:
That sounds good no worries! it was a not a blocker. I'll resolve this for now, yep let's keep it in the struct then.